### PR TITLE
[LUM-834] Better handling of queries timeouts

### PR DESCRIPTION
**Pull request by Segfaultd**
### Introduction

An issue can arise on production where an ICQ isn't picked up & processed by the relayer. If so, it will stay forever pending and since relayers don't provide any way to re-trigger the processing nor forging a custom timeout message, we have to implement a process for this edgy case

### Testing

One test was disabled due to lacking of IBC Testing Framework, and with the code using the client revision height, it would always fail.

Testing is conducted on official Lum Testnet

### CODE REVIEW

File: x/icqueries/types/genesis.go [hunk 1]
```diff
@@ -12,11 +12,6 @@ func DefaultGenesis() *GenesisState {
 
 // Validate performs basic genesis state validation returning an error upon any failure.
 func (gs GenesisState) Validate() error {
-	// Loop through all queries and validate them
-	for _, query := range gs.GetQueries() {
-		if err := query.ValidateBasic(); err != nil {
-			return err
-		}
-	}
+	// TODO: fixme
```
Comment (Ricardo-Remy) [inline x/icqueries/types/genesis.go:15]: I guess we should fix this

File: x/icqueries/types/query.go [hunk 2]
```diff
@@ -1,38 +1,16 @@
...
+func (q *Query) HasTimedOut(currentBlockTime time.Time) bool {
+	return q.TimeoutTimestamp < uint64(currentBlockTime.UnixNano())
+}
 
-	if len(query.GetCallbackId()) <= 0 {
-		return errorsmod.Wrapf(ErrInvalidQuery, "callback id is required")
-	}
-	return nil
+// Description Prints an abbreviated query description for logging purposes
+func (q *Query) Description() string {
```
Comment (Ricardo-Remy) [inline x/icqueries/types/query.go:14]: Maybe I missed it, but I couldn't find anywhere where this function is used? Do we need it?

**Comment by Segfaultd** [comment]
```Sep 22 12:44:26 testnet-validator lumd[368054]: 12:44PM ERR QUERY PROOF VERIFICATION FAILED - QueryId: 3d4cf6801a4500010e26ce5fe1255bfc652027f0540482890a6bf13b2469e0fe, Error: Unable to verify membership proof: chained membership proof failed to verify membership of value: 0A057561746F6D12083832343936383438 in subroot 4FCE629F10F13198725A17669EFAB2C2B681CC09B839D822CC717A75BA46E240 at index 0. Please ensure the path and value are both correct.: invalid proof: icq query response failed module=x/interchainquery
Sep 22 12:44:26 testnet-validator lumd[368054]: 12:44PM ERR QUERY FAILURE - QueryId: 3d4cf6801a4500010e26ce5fe1255bfc652027f0540482890a6bf13b2469e0fe, TTL: 1695388456455986340, BlockTime: 1695386661526348302 module=x/millions
Sep 22 12:44:26 testnet-validator lumd[368054]: 12:44PM INF executed block height=4246782 module=state num_invalid_txs=0 num_valid_txs=1```

**Comment by Segfaultd** [comment]
> > Sep 22 12:44:26 testnet-validator lumd[368054]: 12:44PM ERR QUERY PROOF VERIFICATION FAILED - QueryId: 3d4cf6801a4500010e26ce5fe1255bfc652027f0540482890a6bf13b2469e0fe, Error: Unable to verify membership proof: chained membership proof failed to verify membership of value: 0A057561746F6D12083832343936383438 in subroot 4FCE629F10F13198725A17669EFAB2C2B681CC09B839D822CC717A75BA46E240 at index 0. Please ensure the path and value are both correct.: invalid proof: icq query response failed module=x/interchainquery
> > Sep 22 12:44:26 testnet-validator lumd[368054]: 12:44PM ERR QUERY FAILURE - QueryId: 3d4cf6801a4500010e26ce5fe1255bfc652027f0540482890a6bf13b2469e0fe, TTL: 1695388456455986340, BlockTime: 1695386661526348302 module=x/millions
> > Sep 22 12:44:26 testnet-validator lumd[368054]: 12:44PM INF executed block height=4246782 module=state num_invalid_txs=0 num_valid_txs=1

Fixed by downgrading to IBC v7.2.0

**Review by Ricardo-Remy — APPROVED** [review]

**Pull request merged**